src/KafkaW/Consumer.h

```diff
@@ -9,21 +9,32 @@
 
 namespace KafkaW {
 
-class Inspect;
+class ConsumerInterface {
+public:
+  ConsumerInterface() = default;
+  virtual ~ConsumerInterface() = default;
+  virtual void addTopic(std::string Topic,
+                        const std::chrono::milliseconds &StartTime =
+                            std::chrono::milliseconds{0}) = 0;
+  virtual PollStatus poll() = 0;
+  virtual void dumpCurrentSubscription() = 0;
+  virtual bool topicPresent(const std::string &Topic) = 0;
+  virtual int32_t queryNumberOfPartitions(const std::string &TopicName) = 0;
+};
 
-class Consumer {
+class Consumer : public ConsumerInterface {
 public:
   Consumer(BrokerSettings opt);
   Consumer(Consumer &&) = delete;
   Consumer(Consumer const &) = delete;
-  ~Consumer();
+  virtual ~Consumer();
```

Comment:
Using `override` is preferred instead of `virtual`.

Reply:
This one was slightly weird to me seeing as the destructor is not really overridden in the same way other member functions are. I have changed the code.